rbac/api/utils.py

```diff
@@ -1,15 +1,49 @@
 """Helper utilities for api module."""
 
+import logging
 from django.db import transaction
+from django.shortcuts import get_object_or_404
 from management.utils import account_id_for_tenant
 
 from api.models import Tenant
+from management.role.model import BindingMapping
+from management.tenant_mapping.model import TenantMapping
+from management.workspace.model import Workspace
 
 
+RESOURCE_MODEL_MAPPING = {
+    "workspace": Workspace,
+    "mapping": TenantMapping,
+    "binding": BindingMapping,
+}
+logger = logging.getLogger(__name__)
+
 def populate_tenant_account_id():
     """Populate tenant's account id's."""
     tenants = Tenant.objects.filter(account_id__isnull=True).exclude(tenant_name="public")
     for tenant in tenants:
         with transaction.atomic():
             tenant.account_id = account_id_for_tenant(tenant)
             tenant.save()
+
+
+def migration_resource_deletion(resource, org_id):
+    resource_objs = RESOURCE_MODEL_MAPPING[resource].objects.all()
+    if org_id:
+        tenant = get_object_or_404(Tenant, org_id=org_id)
+        if resource == "binding":
+            resource_objs = resource_objs.filter(role__tenant=tenant)
```

Comment:
This will miss bindings for system roles because those are in the public tenant. Is this intentional?

I think we probably want removing the binding to be based off of the tenant of the resource, however we don't have a foreign key here, and we might not always know the owning Tenant of the bound resource. For now, we actually do, though, since the resource is always a Workspace. Maybe we could just start there? That is, find all workspaces for a tenant, and then delete all the binding mappings for those workspaces (using resource type namespace, resource type name, and resource id fields to filter).

In the long run, we probably won't be able to support removing bindings by org, unless we make it clear it's limited to only certain narrow filter options (like custom roles, or workspace bindings).

It also may be safer for us to just blow away everything, but it probably doesn't hurt to have something org specific in case it is needed. It just has to be used very carefully of course.

Reply:
I had the wrong impression the bindings for system roles have to be kept. I am ignoring the org_id for bindingmapping for now